syntax = "proto3";

package llvm_ml;

message MCSample {
  bool failed = 1;
  uint64 cycles = 2;
  uint64 instructions = 3;
  uint64 uops = 4;
  uint64 cache_misses = 5;
  uint64 context_switches = 6;
}

message MCMetrics {
  uint64 noise_cycles = 1;
  uint64 noise_cache_misses = 2;
  uint64 noise_context_switches = 3;
  uint64 noise_num_runs = 4;
  uint64 workload_cycles = 5;
  uint64 workload_cache_misses = 6;
  uint64 workload_context_switches = 7;
  uint64 workload_num_runs = 8;
  uint64 measured_cycles = 9;
  uint64 measured_num_runs = 10;
  optional string source = 11;
  repeated MCSample noise_samples = 12;
  repeated MCSample workload_samples = 13;
}
